### PR TITLE
Updated css for print media

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
https://github.com/mitodl/mitx-theme/issues/23

This PR modifies the print media of edx platform.

@pdpinch 

screen shot 2017-08-22 at 4 22 32 pm

### CODE REVIEW

**Comment by pdpinch** [comment]
LGTM. 

@amir-qayyum-khan please merge this into the master branch, and then also into ficus and ficus-draft branches. 

Then, when convenient, @blarghmatey can release to the QA stack for @jolyonb to review

**Comment by amir-qayyum-khan** [comment]
Done @pdpinch

**Comment by blarghmatey** [comment]
@pdpinch @jolyonb this has been released to QA (https://mitx-qa-draft.mitx.mit.edu and https://mitx-qa.mitx.mit.edu)

**Comment by jolyonb** [comment]
I checked out printing on the mitx-qa server. It looks like some of the changes work, but some are being overridden somewhere. Because the script is interpreted with SASS and then the load order is a bit weird, things can work out differently to just injecting css via greasemonkey (which was how I came up with the css to begin with). I'll go through the code and put individual comments on lines that need changing.

File: lms/static/sass/_overrides.scss [hunk 1]
```diff
@@ -68,3 +68,58 @@ ol.nav-global>li>a {
   font-style: italic !important;
   font-size: 15px !important;
 }
+
+@media print {
+  .course-index {
+      display: none;
```
Comment (jolyonb) [inline lms/static/sass/_overrides.scss]: Needs !important. Being overridden by load order of CSS files.

File: lms/static/sass/_overrides.scss [hunk 2]
```diff
@@ -68,3 +68,58 @@ ol.nav-global>li>a {
...
+  #global-navigation {
+      border-bottom: 1px solid #C8C8C8 !important;
+  }
+
+  footer#footer-edx-v3 {
+      display: none;
+  }
+
+  .container > div {
+      border: none;
```
Comment (jolyonb) [inline lms/static/sass/_overrides.scss]: Needs !important. Being overridden by load order of CSS files.

File: lms/static/sass/_overrides.scss [hunk 3]
```diff
@@ -68,3 +68,58 @@ ol.nav-global>li>a {
...
+  }
+
+  .xmodule_display.xmodule_CapaModule div.problem .action {
+      display: none;
```
Comment (jolyonb) [inline lms/static/sass/_overrides.scss]: Needs !important. Being overridden by load order of CSS files.

File: lms/static/sass/_overrides.scss [hunk 4]
```diff
@@ -68,3 +68,58 @@ ol.nav-global>li>a {
...
+  }
+
+  .course-content .bookmark-button-wrapper {
+      display: none;
+  }
+
+  .container .courseware-bookmarks-button {
+      display: none;
```
Comment (jolyonb) [inline lms/static/sass/_overrides.scss]: Unnecessary when .course-index is hidden.

File: lms/static/sass/_overrides.scss [hunk 5]
```diff
@@ -68,3 +68,58 @@ ol.nav-global>li>a {
...
+      margin-top: -12px;
+      margin-bottom: 2px;
+  }
+
+  a, a:visited {
+      text-decoration: underline !important;
+  }
+
+  .xmodule_display.xmodule_CapaModule div.problem-progress {
+      display: none;
```
Comment (jolyonb) [inline lms/static/sass/_overrides.scss]: Needs !important. Being overridden by load order of CSS files.

File: lms/static/sass/_overrides.scss [hunk 6]
```diff
@@ -68,3 +68,58 @@ ol.nav-global>li>a {
...
+  }
+
+  .container .courseware-search-bar {
+      display: none;
+  }
+
+  .course-wrapper .course-content .path, .course-wrapper .courseware-results-wrapper .path {
+      margin-top: -12px;
+      margin-bottom: 2px;
```
Comment (jolyonb) [inline lms/static/sass/_overrides.scss]: Both of these lines needs !important. Being overridden by load order of CSS files.

File: lms/static/sass/_overrides.scss [hunk 7]
```diff
@@ -68,3 +68,58 @@ ol.nav-global>li>a {
...
+  a, a:visited {
+      text-decoration: underline !important;
+  }
+
+  .xmodule_display.xmodule_CapaModule div.problem-progress {
+      display: none;
+  }
+
+  .course-wrapper .course-content .vert-mod .vert>.xblock-student_view {
+      padding: none;
```
Comment (jolyonb) [inline lms/static/sass/_overrides.scss]: Should be:
padding: 0 !important;
My fault on the semantic error. Again, overridden by load order of CSS.

**Comment by jolyonb** [comment]
I've finished marking up the commit here, based on my review of the QA server.

**Comment by pdpinch** [comment]
Thanks Jolyon. @amir-qayyum-khan can you refactor this based on Jolyon's input? 

I don't know if there are any options to avoid the "important" flags. I'm fine with them for the theme, but we'll have to do something more fundamental when we go to edX with these changes.

**Comment by jolyonb** [comment]
@pdpinch I agree - I prefer to avoid them if possible. I had though that being in a "@media print" made the tagging more specific, and hence would override, but it appears not to be happening. Happy to discuss for when we get to the edX part of things. Indeed, it even looks like a bunch of these stylings are already in the edX codebase (from when I put them there previously), but they're being overridden. We may just need to increase their specificity. I'd love to get it working properly in edX, so that we don't need any !importants in the theme at all!

**Pull request merged**